mParticle-Apple-SDK/Utils/NSDictionary+MPCaseInsensitive.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  NSDictionary+MPCaseInsensitive.swift
+//  mParticle-Apple-SDK
+//
+//  Created by Brandon Stalnaker on 1/29/25.
+//
+
+import Foundation
+public extension NSDictionary {
+
+    @objc func caseInsensitiveKey(_ key: String) -> String? {
+        var localKey = key
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                localKey = stringKey
+            }
+        }
+        return localKey
+    }
+
+    @objc func value(forCaseInsensitiveKey key: String) -> Any? {
+        var value : Any?
+        
+        self.allKeys.forEach { keyValue in
+            if let stringKey = keyValue as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
+                value = self[stringKey]
+            }
+        }
+        
+        return value
```

Comment:
Same here. Also caqn be slightly simplified since with a `for in` loop you can get the key and value directly:

```suggestion
        for (aKey, value) in self {
            if let stringKey = aKey as? String, stringKey.caseInsensitiveCompare(key) == .orderedSame {
                return value
            }
        }
        return nil
```